contracts/route/route.go

```diff
@@ -22,6 +23,8 @@ type Route interface {
 	RunTLSWithCert(host, certFile, keyFile string) error
 	// ServeHTTP serves HTTP requests.
 	ServeHTTP(writer http.ResponseWriter, request *http.Request)
+	// Shutdown gracefully shuts down the serve.If the provided context expires before the shutdown is complete, Shutdown returns the context's error.
+	Shutdown(ctx context.Context) error
```

Comment:
Users may use `Run`, `RunTLS`, `RunTLSWithCert` at the same time, so I'm thinking if these three methods should return the server instance directly, instead of adding a single `Shutdown` method. Or the `Shutdown` method can deal with a `server` slice, and shut down them simultaneously, but this way is a bit inflexible. Any thoughts, please?

Reply:
I think the server instance itself should not consider the situation of multiple servers, this should be managed at the upper level; It may be necessary to handle multiple server instances from the perspective of containers.

In addition, implementing a server that supports gracefully quit is generally like this:
```golang
func main() {
    srv := facades.Route()

    go func() {
        if err := srv.Run(); err != nil {
	        log.Fatal(err)
        }
    }()
    
    quit := make(chan os.Signal)
    signal.Notify(quit, syscall.SIGINT, syscall.SIGTERM)
    <-quit
    log.Println("Shutdown Server ...")
    
    ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
    defer cancel()
    if err := srv.Shutdown(ctx); err != nil {
    log.Fatal("Server Shutdown:", err)
    }
    
    log.Println("Server exiting")
}
```
So, whether `Run` returns a server instance or not is not important, as they are generally not executed in the same goroutine

Reply:
Good idea, if so, we just need to change `Singleton` to `Bind` and add the `Shutdown` method.

<img width="752" alt="image" src="https://github.com/user-attachments/assets/a33058e9-dd77-4923-9a29-267e01181fd0">